core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -71,7 +71,7 @@ class KafkaApis(val requestChannel: RequestChannel,
   def handle(request: RequestChannel.Request) {
     try {
       trace("Handling request:%s from connection %s;securityProtocol:%s,principal:%s".
-        format(request.requestObj, request.connectionId, request.securityProtocol, request.session.principal))
+        format(Option(request.header).getOrElse(request.requestObj), request.connectionId, request.securityProtocol, request.session.principal))
```

Comment:
Sorry, looking a bit deeper, I wonder if we should call the following method:

```scala
private def requestDesc(details: Boolean): String = {
      if (requestObj != null)
        requestObj.describe(details)
      else
        header.toString + " -- " + body.toString
    }
```

It's private, but maybe we should change that. What do you think?